lib/truffle/truffle/patching.rb

```diff
@@ -1,140 +1,69 @@
-require 'pathname'
-
 module Truffle::Patching
-  DIR     = Pathname(Truffle::Boot.ruby_home).join("lib/patches")
-  # Allowed operations are: :before, :after, :instead, :ignore
-  PATCHES = { "bundler"                                 => :before,
-              "bundler/version"                         => :after,
-              "bundler/cli/exec"                        => :after,
-              "bundler/compact_index_client/updater"    => :after,
-              "bundler/current_ruby"                    => :after,
-              "bundler/dependency"                      => :after,
-              "bundler/fetcher/downloader"              => :after,
-              "bundler/fetcher/compact_index"           => :after,
-              "bundler/ruby_version"                    => :after,
-              "bundler/source/rubygems"                 => :after,
-
-              "rubygems"                                => :before,
-              "rubygems/ext"                            => :after,
-              "rubygems/package"                        => :after,
-              "rubygems/remote_fetcher"                 => :after,
-              "rubygems/request"                        => :after,
-              "rubygems/request_set/gem_dependency_api" => :after,
-
-              "rspec/support"                           => :after,
-
-              "unf/normalizer_cruby"                    => :before,
-  }
-
-  PATCHES.each do |file, _|
-    unless DIR.join(file).sub_ext('.rb').exist?
-      raise "file #{file} is missing in #{DIR}"
+  extend self
+
+  def patches
+    @patches ||= begin
+      require 'pathname'
+      Pathname.glob(dir.join('*')).reduce({}) do |patches, path|
+        next patches unless path.directory?
+        patches.update path.basename.to_s => path
+      end
```

Comment:
This file uses only pathname API, even though it's not really necessary in this particular method I don't think mixing 2 APIs in one file is an improvement. I prefer functional style. I respect your code written in imperative style and I am not asking you to rewrite it to FP style, please do the same. In the end, there is little difference and I don't think it makes sense to ask each other to rewrite it in one way or the other. However `each_with_object` will allow to put the condition on one line.